castled-notifications/src/main/java/io/castled/android/notifications/inbox/viewmodel/InboxRepository.kt

```diff
@@ -0,0 +1,136 @@
+import io.castled.android.notifications.logger.CastledLogger
+import io.castled.android.notifications.logger.LogTags
+import io.castled.android.notifications.network.CastledRetrofitClient
+import io.castled.android.notifications.store.CastledDbBuilder
+import io.castled.android.notifications.store.CastledSharedStore
+import io.castled.android.notifications.store.models.AppInbox
+import io.castled.android.notifications.workmanager.CastledNetworkWorkManager
+import io.castled.android.notifications.workmanager.models.CastledInboxEventRequest
+import retrofit2.Response
+import java.io.IOException
+
+internal class InboxRepository(context: Context) {
+
+    internal val inboxDao = CastledDbBuilder.getDbInstance(context).inboxDao()
+    private val logger = CastledLogger.getInstance(LogTags.INBOX_REPOSITORY)
+    private val inboxApi = CastledRetrofitClient.create(InboxApi::class.java)
+    private val networkWorkManager = CastledNetworkWorkManager.getInstance(context)
+    internal val cachedInboxItems = mutableListOf<AppInbox>()
+    internal suspend fun refreshInbox() {
+        val liveInboxResponse = fetchLiveInbox() ?: return
+        val liveInboxItems = liveInboxResponse.map { it.toInbox() }
+        val cachedInboxItemsMapById = cachedInboxItems.associateBy { it.messageId }
+        val liveInInboxItemsMapById = liveInboxItems.associateBy { it.messageId }
+        val expiredInboxItems =
+            cachedInboxItems.filterNot { liveInInboxItemsMapById.containsKey(it.messageId) }
+        val newInboxItems =
+            liveInboxItems.filterNot { cachedInboxItemsMapById.containsKey(it.messageId) }
+        insertInboxIntoDb(newInboxItems)
+        deleteDbInbox(expiredInboxItems)
+    }
+
+    private suspend fun insertInboxIntoDb(inboxItems: List<AppInbox>): LongArray {
+        return inboxDao.dbInsertInbox(inboxItems)
+    }
+
+    private suspend fun deleteDbInbox(inboxItems: List<AppInbox>): Int {
+        return inboxDao.dbDeleteAllInboxItems(inboxItems)
+    }
+
+    internal fun deleteInboxItem(inbox: AppInbox) {
+        inboxDao.delete(inbox)
+    }
+
+    private suspend fun fetchLiveInbox(): List<CastledInboxItem>? {
+        try {
+            val response = inboxApi.fetchInboxItems(
+                CastledSharedStore.getAppId(), CastledSharedStore.getUserId()
+            )
+            return if (response.isSuccessful) {
+                response.body()
+            } else {
+                // Handle API errors (e.g., 4xx or 5xx status codes)
+                val errorMessage = response.errorBody()?.string() ?: "Unknown error"
+                logger.error(errorMessage)
+                null
+            }
+        } catch (e: IOException) {
+            logger.error("Network error!", e)
+        } catch (e: Exception) {
+            logger.error("Unknown error!", e)
+        }
+        return null
+    }
+
+    internal suspend fun deleteInboxItem(
+        inboxItem: CastledInboxItem, completion: (Boolean, String) -> Unit
```

Comment:
better to name callbacks using `onSomethingHappened` format to be consistent with other parts of the code. So here it would be  `onInboxItemDeleted`. Also it is cleaner to split the callbacks for success and failures. 
so you will have a `onSuccess` and `onError` callbacks and avoid passing boolean flags. it will help with code readability down the line